release/howto-update-conan-package.md

```diff
@@ -0,0 +1,187 @@
+# How-to Guide: Update Conan Package
+
+Updating the Conan packages is relatively involved and typically takes weeks for
+the PRs to go through reviews. Check with coryan@ or dbolduc@ before embarking
+on this quest.
+
+This package manager uses CMake, but ignores the library and target definitions
+exported by our CMake files. It reimplements the library definitions and
+dependencies in Python code. The recipes do this for every package (gRPC,
+Protobuf, Boost, etc.), not just for `google-cloud-cpp`. For most packages,
+where there are only a handful of libraries, this is trivial. But
+`google-cloud-cpp` has about 240 libraries (including the `*_protos` libraries),
+so we use a helper script to automatically generate the Python data structures
+that describe the libraries in each of our versions.
+
+This package manager requires patches to our code. These patches need to be
+updated on each release. Package management systems tend to apply patches with
+very strict settings, so even small changes around the patches break them.
+Sometimes one can use `patch(1)` manually, with looser settings, and use that to
+update the patches.
+
+\[PR#17988\] is probably a good example of the changes you will need to make.
+
+## Prepare your environment
+
+Create a fork of
+[conan-center-index](https://github.com/conan-io/conan-center-index.git)
+
+Clone the fork:
+
+```shell
+git clone git@github.com:${GITHUB_USERNAME}/conan-center-index
+cd conan-center-index
+```
+
+## Verify your environment is able to build the current Conan package
+
+We will use `TAG=2.15.1` and `UTAG=2_15_1` and `LATEST=2.12.0` in these
+examples. `LATEST` represents the latest version in Conan, `TAG` represents the
+version we want to add to Conan. It is common to skip several `google-cloud-cpp`
+releases as it may take months to complete one update.
+
+If you have not used Conan in a while, you may want to remove any stale
+directories:
+
+```shell
+rm -fr ~/.conan2
+rm -fr ~/.venv/conan2
+```
+
+Create an isolated Python environment and then install Conan and the hooks for
+Conan package development:
+
+```shell
+python3 -m venv ~/.venv/conan2
+# Activate the environment
+. ~/.venv/conan2/bin/activate
+# Install conan, it is a Python package
```

Comment:
Perhaps _"Install conan (it is a Python package)"_?

Reply:
Done